cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -161,10 +160,14 @@ private void checkServicesWithRetry(Callable<Boolean> callable,
         }
         TimeUnit.SECONDS.sleep(1);
       } catch (Throwable e) {
-        if (!(e.getCause() instanceof InterruptedException || e.getCause() instanceof IOException)) {
-          // we want to throw UnauthorizedException while suppress and retry on Interrupted or IOException
+        // we want to throw UnauthorizedException while suppress and retry on Interrupted or IOException
```

Comment:
Maybe instead of catching Throwable, use -
```java
} catch (InterruptedException | IOException e) {
 // suppress InterruptedException and IOException
}
```